proposals/utils/statistics_utils.py

```diff
@@ -0,0 +1,265 @@
+
+
+def get_total_short_route_proposals(data: QuerySet) -> int:
+    """Calculates the total amount of proposals in the data that were reviewed
+    using the short route.
+
+    :param data: QuerySet[Proposal], the proposals to calculate this
+    statistic for
+    :return: the num of short route proposals in the given QuerySet
+    :rtype: int
+    """
+    return get_qs_for_short_route_reviews(data).count()
+
+
+def get_total_long_route_proposals(data: QuerySet) -> int:
+    """Calculates the total amount of proposals in the data that were reviewed
+    using the long route.
+
+    :param data: QuerySet[Proposal], the proposals to calculate this
+    statistic for
+    :return: the num of long route proposals in the given QuerySet
+    :rtype: int
+    """
+    return get_qs_for_short_route_reviews(data).count()
+
+
+def get_total_students(data: QuerySet) -> dict:
+    """Calculates a dict of all Relations that are deemed to be students,
+    and a corresponding number of occurrences in the supplied Proposals.
+
+    :param data: QuerySet[Proposal], the proposals to calculate this
+    statistic for
+    :return: A dict of (relation, count) pairs
+    :rtype: dict
+    """
+    relations = Relation.objects.filter(check_in_course=True)
+    totals = defaultdict(int)
+    for relation in relations:
+        totals[relation.description_en] = data.filter(relation=relation).count()
+
+    return totals
+
+
+def get_total_turnaround_time(review_data: QuerySet) -> Counter:
+    """Calculates the amount of days it took to conclude a review, and puts
+    them in a counter, in which the left-hand side represents the number of
+    days and the right-hand side represents the number of reviews concluded
+    in that amount of days
+
+    :param review_data: QuerySet[Review], the reviews to calculate this
+    statistic for
+    :return: A Counter of (days, count) pairs
+    :rtype: Counter
+    """
+    return Counter(
+        [(x.date_end - x.date_start).days for x in review_data]
+    )
+
+
+def get_average_turnaround_time(review_data: QuerySet) -> float:
+    """Calculates the average time it took to conclude all reviews in the
+    given dataset
+
+    :param review_data: QuerySet[Review], the reviews to calculate this
+    statistic for
+    :return: the average turnaround time on a review, expressed in days (
+    using a float to represent partial days)
+    :rtype: float
+    """
+    total = review_data.count()
+
+    # Protect against division by 0
+    if total == 0:
+        return 0
+
+    # First calculate the delta in seconds and then divide it by 60 to get
+    # hours and then by 24 to get a day representation in floats.
+    # Sidenote; PYTHON WHY ARE MY ONLY OPTIONS MICROSECONDS, SECONDS AND DAYS
+    # We don't sum the num of days, as some reviews are closed in less than a
+    # day which would cause the calculation to ignore that review
+    return sum(
+        [(x.date_end - x.date_start).seconds / 60 / 24 for x in review_data]
```

Comment:
Sixty seconds in an hour? Wait a minute...

Reply:
Oops lol

Reply:
The comment above still reflects the old code, but otherwise this can merge IMO